jax/_src/numpy/lax_numpy.py

```diff
@@ -442,8 +442,39 @@ def fmax(x1: ArrayLike, x2: ArrayLike) -> Array:
 def issubdtype(arg1: DTypeLike, arg2: DTypeLike) -> bool:
   return dtypes.issubdtype(arg1, arg2)
 
-@util.implements(np.isscalar)
 def isscalar(element: Any) -> bool:
+  """Check if an element is a scalar.
+
+  JAX implementation of :func:`numpy.isscalar`.
+
+  The JAX version has a few differences from the NumPy version:
+
+  - In NumPy, scalars and zero-dimensional arrays are distinct.
+  - JAX treats zero-dimensional arrays as scalars only if they originate from Python scalars.
+  - NumPy scalar types like ``np.float32`` or ``np.complex64`` are treated as scalars, while JAX types
```

Comment:
Similarly here, I think this will be unhelpful to most people. Most users in my experience, even established researchers who have been using NumPy for years, will think that `np.float32` is a dtype, and won't appreciate that it's a scalar type that can be instantiated to create a scalar object.